### PR TITLE
Release 3.4.2

### PR DESCRIPTION
**Pull request by unconst**
[Fix] promo change to axon and dendrite https://github.com/opentensor/bittensor/pull/981
[Feature] no version checking flag https://github.com/opentensor/bittensor/pull/974
[Fix] add perpet hash rate and adjust alpha https://github.com/opentensor/bittensor/pull/960
[Fix] stake conversion issue https://github.com/opentensor/bittensor/pull/958
[Feature] Dendrite asyncio https://github.com/opentensor/bittensor/pull/967
[Fix] Bit 590 backward fix https://github.com/opentensor/bittensor/pull/957
[Feature] No set weights https://github.com/opentensor/bittensor/pull/959
[Feature] Improve dataloader performance https://github.com/opentensor/bittensor/pull/950
[Fix] Remove locals from cli and bittensor common https://github.com/opentensor/bittensor/pull/947
[Fix] Minor fixes https://github.com/opentensor/bittensor/pull/955

### CODE REVIEW

**Review by shibshib — APPROVED** [review]

File: bittensor/__init__.py [hunk 1]
```diff
@@ -16,16 +16,29 @@
...
 from rich.console import Console
+from rich.traceback import install
 from prometheus_client import Info
 
+import nest_asyncio
+nest_asyncio.apply()
+
 # Bittensor code and protocol version.
-__version__ = '3.4.1'
+__version__ = '3.4.2'
```
Comment (eduardogr) [inline bittensor/__init__.py:26]: remember also to update this version: https://github.com/opentensor/bittensor/blob/Release_3.4.2/VERSION

Next times you can use the script: https://github.com/opentensor/bittensor/blob/Release_3.4.2/scripts/release/update_version.sh

File: bittensor/_prometheus/__init__.py [hunk 2]
```diff
@@ -126,7 +126,7 @@ def add_defaults(cls, defaults):
         defaults.prometheus = bittensor.Config()
         # Default the prometheus port to axon.port - 1000
         defaults.prometheus.port = os.getenv('BT_PROMETHEUS_PORT') if os.getenv('BT_PROMETHEUS_PORT') != None else 7091
-        defaults.prometheus.level = os.getenv('BT_PROMETHEUS_LEVEL') if os.getenv('BT_PROMETHEUS_LEVEL') != None else bittensor.prometheus.level.OFF.value
+        defaults.prometheus.level = os.getenv('BT_PROMETHEUS_LEVEL') if os.getenv('BT_PROMETHEUS_LEVEL') != None else bittensor.prometheus.level.INFO.value
```
Comment (Eugene-hu) [inline bittensor/_prometheus/__init__.py:129]: This means that all servers and validators will turn on prometheus by default, correct?

**Review by Eugene-hu — APPROVED** [review]
LGTM

File: bittensor/_prometheus/__init__.py (hunk 2 above)
Comment (eduardogr) [reply bittensor/_prometheus/__init__.py:129]: yes, that's correct

**Comment by unconst** [comment]
Release scheduled for monday

**Comment by unconst** [comment]
Requires a push to nobunaga before merging to master

**Comment by camfairchild** [comment]
this should close #973 when merged

**Pull request merged**